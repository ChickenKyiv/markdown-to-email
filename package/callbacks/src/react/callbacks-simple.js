import { replaceReactWrapper, newLine } from '../helpers';

const hr = `${newLine}<hr />`;
const empty = '';

function previewText(text, content) {
  const config = {
    content,
  };

  //wouldnt work well
  this.previewText = true;



  return replaceReactWrapper('previewText', config, 'body');
}

function code(text, content) {
  const config = {
    content,
  };



  const result = replaceReactWrapper('code', config);
  return result;
}

function del(text, content) {
  const config = {
    content,
  };



  const result = replaceReactWrapper('del', config);
  return result;
}

function q(text, content) {
  const config = {
    content,
  };



  const result = replaceReactWrapper('q', config);
  return result;
}

function italic(text, left, _, content, right) {
  const config = {
    content,
  };


  const result = replaceReactWrapper('italic', config);
  return result;
}

function strong(text, doubleAsterix, content, asterix) {
  const config = {
    content: `${content + asterix}`,
  };



  const result = replaceReactWrapper('strong', config);
  return result;
}

function link(text, title, href) {
  const config = {
    content: title.trim(),
    href: href.trim(),
  };



  const result = replaceReactWrapper('link', config);
  return result;
}

function blockquote(text, tmp, item) {
  const config = {
    content: `${newLine}${item.trim()}`,
  };


  const result = replaceReactWrapper('blockquote', config);
  return result;
}

function mem(text, src, href, altText) {
  const config = {
    src: src.trim(),
    altText: altText.trim(),
    href: href.trim(),
  };



  const result = replaceReactWrapper('image', config);
  return result;
}

function header(text, chars, content) {
  const config = {
    content: content.trim(),
  };

  const titleType = ['mainTitle', 'subtitle', 'heading'];



  const result = newLine + 
    replaceReactWrapper(titleType[chars.length - 1], config);

  return result;
}

function separator() {
  const config = {};


  
  const result = `${newLine}${replaceReactWrapper(
    'separator',
    config,
  )}${newLine}`;

  return result;
}

export {
  strong,
  link,
  blockquote,
  mem,
  header,

  italic,
  del,
  q,
  code,
  hr,
  empty,
  previewText,
  separator,
};
