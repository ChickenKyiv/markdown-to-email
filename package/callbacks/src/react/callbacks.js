import { replaceReactWrapper, newLine } from '../helpers';

/// function is not working as planned

function paragraphWrapper(text, line) {
  const trimmed = line.trim();
  if (/^<\/?(ul|ol|li|h|p|bl)/i.test(trimmed)) {
    // @TODO move out this regex into constants file.
    return newLine + line + newLine;
  }

  const config = {
    content: trimmed,
  };


  const replaced = replaceReactWrapper(
    'paragraph', 
    config
  );

  const result = newLine + replaced + newLine;
  // console.log(config);

  return result;
}
/// !end function is not working as planned

function image(text, alt, srcWithTooltip) {
  const src = srcWithTooltip.trim().replace(/\"image_tooltip\"/, '');

  const config = {
    src: src.trim(),
    altText: alt,
  };

  // not sure if it will work fine
  this.warnings.images++;



  const replaced = replaceReactWrapper(
    'image', 
    config
  );

  
  return replaced;
}

function br(text, newLines) {
  const arrNewLines = newLines.match(new RegExp(newLine, 'g'));

  // @TODO well, it's not good. can be improved with lodash
  // @TODO I REALLY HATE THIS LINE
  const result = arrNewLines.reduce(
    (acc, current, index) => (index > 0 ? `${acc}<br/>${current}` : current), '',
  );

  return result;
}

function sponsorship(text) {
  const regex = /\[(.*?)\]/g;
  const regex2 = /[\[\]]/g;

  const [content, href, src] = text
    .match(regex)
    .map((match) => match.replace(regex2, ''));

  const config = {
    src: src.trim(),
    href: href.trim(),
    content: content.trim(),

    folder: 'body'
  };

  // @TODO nope, not good
  this.errors.sponsorshipTop
    ? (this.errors.sponsorshipBottom = true)
    : (this.errors.sponsorshipTop = true);


    const replaced = replaceReactWrapper(
      'sponsor', 
      config,
      'body'
    );  

    return replaced;
}

// function tag_loop(){
//     var arr = {
//         header,
//         content,
//         footer,
//        social,
//         unsub
//     }
//     var completed_HTML = '';
//     _.each(arr,{

//         completed_HTML += element;

//     });
// }

export {
  paragraphWrapper,
  image,
  sponsorship,
  br,
  newLine
};
