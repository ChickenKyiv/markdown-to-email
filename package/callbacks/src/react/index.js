import {
  strong,
  link,
  blockquote,
  mem,
  header,
  italic,
  del,
  q,
  code,
  hr,
  empty,
  previewText,
  separator
} from './callbacks-simpleMDReact';

import {
  image,
  paragraphWrapper,
  sponsorship,
  br
} from './callbacksMDReact';

import {
  ulList,
  olList
} from './callbacks-lists-react';



export default {
    strong,
    link,
    blockquote,
    mem,
    header,
    italic,
    del,
    q,
    code,
    hr,
    empty,

    ulList,
    olList,

    image,
    paragraphWrapper,
    sponsorship,
    br,

    separator
};
