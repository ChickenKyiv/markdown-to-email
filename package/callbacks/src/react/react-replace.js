import {
newLine
} from '../helpers';

import {
  strong,
  link,
  blockquote,
  mem,
  header,
  italic,
  del,
  q,
  code,
  hr,
  empty,
  previewText,
  separator
} from './callbacks-simpleMDReact';

import {
  image,
  paragraphWrapper,
  sponsorship,
  br
} from './callbacksMDReact';

import {
  ulList,
  olList
} from './callbacks-lists-react';

import * as constants from 'atherdon-newsletter-constants';

class ReactReplacer {
    constructor(constants){
        this._constants = constants;

        this.replaceMDBinded = () => {};
        this.replaceMDBindedPreviewText = () => {};

        this.previewText = function () {
            this.replaceMDBinded(this._constants.REGEXP_PREVIEW_TEXT, previewText);
        };

        this.comments = function () {
            this.replaceMDBinded(this._constants.REGEXP_HTML_COMMENTS, empty);
        };

        this.strong = function () {
            this.replaceMDBinded(this._constants.REGEXP_STRONG, strong);
        };

        this.italic = function () {
            this.replaceMDBinded(this._constants.REGEXP_EM, italic);
        };

        this.header = function () {
            this.replaceMDBinded(this._constants.REGEXP_HEADER, header);
        };

        this.image = function () {
            this.replaceMDBinded(this._constants.REGEXP_IMAGE, image);
        };

        this.link = function () {
            this.replaceMDBinded(this._constants.REGEXP_LINK, link);
        };

        this.del = function () {
            this.replaceMDBinded(this._constants.REGEXP_DEL, del);
        };

        this.q = function () {
            this.replaceMDBinded(this._constants.REGEXP_Q, q);
        };

        this.code = function () {
            this.replaceMDBinded(this._constants.REGEXP_CODE, code);
        };

        this.ulList = function () {
            this.replaceMDBinded(this._constants.REGEXP_UL_LIST, ulList);
        };

        this.olList = function () {
            this.replaceMDBinded(this._constants.REGEXP_OL_LIST, olList);
        };

        this.blockQuote = function () {
            this.replaceMDBinded(this._constants.REGEXP_BLOCKQUOTE, blockquote);
        };

        this.hr = function () {
            this.replaceMDBinded(this._constants.REGEXP_HR, hr);
        };

        this.paragraph = function () {
            this.replaceMDBinded(this._constants.REGEXP_PARAGRAPH, paragraphWrapper);
        };

        this.emptyUl = function () {
            this.replaceMDBinded(this._constants.REGEXP_EMPTY_UL, empty);
        };

        this.emptyOl = function () {
            this.replaceMDBinded(this._constants.REGEXP_EMPTY_OL, empty);
        };

        this.emptyBlockQuote = function () {
            this.replaceMDBinded(this._constants.REGEXP_EMPTY_BLOCKQUOTE, newLine);
        };

        this.br = function () {
            this.replaceMDBinded(this._constants.REGEXP_BR, br);
        };

        this.sponsorship = function () {
            this.replaceMDBinded(this._constants.REGEXP_SPONSORSHIP, sponsorship);
        };

        this.mem = function () {
            this.replaceMDBinded(this._constants.REGEXP_MEM, mem);
        };

        this.separator = function () {
            this.replaceMDBinded(this._constants.REGEXP_SEPARATOR, separator);
        };

    }
}

export default new ReactReplacer(constants);
