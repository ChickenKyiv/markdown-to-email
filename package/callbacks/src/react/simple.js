import { replaceReactWrapper, newLine } from '../helpers';

const hr = `${newLine}<hr />`;
const empty = '';

function previewText(text, content) {
  const config = {
    content: content,
    folder: 'body'
  };



  //wouldnt work well
  this.previewText = true;


  const replaced = replaceReactWrapper(
    'previewText', 
    config,
    'body'
  );
  
  return replaced;
}

function code(text, content) {
  const config = {
    content,
  };

  const replaced = replaceReactWrapper(
    'code', 
    config
  );

  return replaced;
}

function del(text, content) {
  const config = {
    content,
  };

  const replaced = replaceReactWrapper(
    'del', 
    config
  );

  return replaced;
}

function q(text, content) {
  const config = {
    content,
  };

  const replaced = replaceReactWrapper(
    'q', 
    config
  );

  return replaced;
}

function italic(text, left, _, content, right) {
  const config = {
    content,
  };

  const replaced = replaceReactWrapper(
    'italic', 
    config
  );

  return replaced;
}

function strong(text, doubleAsterix, content, asterix) {
  const config = {
    content: `${content + asterix}`,
  };

  const replaced = replaceReactWrapper(
    'strong', 
    config
  );

  return replaced;
}

function link(text, title, href) {
  const config = {
    content: title.trim(),
    href: href.trim(),
  };

  const replaced = replaceReactWrapper(
    'link', 
    config
  );

  return replaced;
}

function blockquote(text, tmp, item) {
  const config = {
    content: `${newLine}${item.trim()}`,
  };

  const replaced = replaceReactWrapper(
    'blockquote', 
    config
  );

  return replaced;
}

function mem(text, src, href, altText) {
  const config = {
    src: src.trim(),
    altText: altText.trim(),
    href: href.trim(),
  };

  const replaced = replaceReactWrapper(
    'image', 
    config
  );

  return replaced;
}

function header(text, chars, content) {
  const config = {
    content: content.trim(),
  };

  const titleType = [
    'mainTitle', 
    'subtitle',
     'heading'
    ];

    const replaced = replaceReactWrapper(
      titleType[chars.length - 1], 
      config
    );


  return newLine + replaced;
}

function separator() {
  const config = {};


  const replaced = replaceReactWrapper(
    'separator', 
    config
  );


  return newLine + replaced + newLine;
}

export {
  strong,
  link,
  blockquote,
  mem,
  header,

  italic,
  del,
  q,
  code,
  hr,
  empty,
  previewText,
  separator,
};
