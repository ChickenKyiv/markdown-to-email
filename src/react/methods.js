const { write } = require('markup-generator');

const reactComponent = `
import React from "react";

const Content = () => {
  return (
    <>
      {content}
    </>
  );
};

export default Content;
`;

function reactComponentReplace(content){
  return reactComponent.replace(/{content}/g, content);
}

function writeReactComponent(fileName, content, dir = 'generated', message) {

  const result = reactComponentReplace(content);
  write(fileName, result, dir, message);
}
