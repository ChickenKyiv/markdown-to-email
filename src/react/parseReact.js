const {
  ReactReplacer, 
  replaceMarkdown, 
  replaceMarkdownPreviewText
} = require('atherdon-callbacks');

const { stateInit } = require('markup-generator');

// ReactReplacer

// @todo update this method. I'm sure it can be improved.
function parseMDReact(source, isFull) {
  const state = stateInit(source);

  // CAN WE MOVE OUT CALLBACKS INTO REPLACER?????
  ReactReplacer.replaceMDBinded = replaceMarkdown.bind(state);
  
  // I think we didnt use it quite often
  // ReactReplacer.replaceMDBindedPreviewText = replaceMarkdownPreviewText.bind(state);

  // NOTE DON'T CHANGE ORDER OF FUNCTION CALLS
  if (isFull) {
    ReactReplacer.previewText();
  }


  ReactReplacer.comments();
  ReactReplacer.typography();

  ReactReplacer.ulList();
  // TODO UPDATE OLLIST CALLBACK, IT'S NOT WORKING
  //   ReactReplacer.olList()

  ReactReplacer.blockQuote();
  ReactReplacer.hr();
  ReactReplacer.paragraph();

  // TODO CHECK CASES WHEN CONTENT CAN HAVE EMPTY LISTS OR BLOCKQUOTE
  //   ReactReplacer.emptyUl();
  //   ReactReplacer.emptyOl();
  //   ReactReplacer.emptyBlockQuote();

  
  ReactReplacer.br();
  ReactReplacer.sponsorship();
  ReactReplacer.mem();
  ReactReplacer.separator();


  // console.log( state )

  return state;
}

function parseMDReactFullThing(params) {
  const { source } = params;

  return parseMDReact(source, true);
}

module.exports = { parseMDReact, parseMDReactFullThing };
