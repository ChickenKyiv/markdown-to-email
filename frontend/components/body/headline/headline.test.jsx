import React from "react"
import {create} from "react-test-renderer"
import HeadLine from "./index"
import ReactDOM from "react-dom";
import {cleanup} from "@testing-library/react";

afterEach(() => {
    cleanup()
});

describe("headline/index component", () => {

    test("renders without crashing", () => {
        const div = document.createElement('div')
        ReactDOM.render(<HeadLine />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    test('renders correctly', () => {
        const tree = create(<HeadLine><h1>Some text</h1></HeadLine>)
        expect(tree.toJSON()).toMatchSnapshot();
    });

    test("compoents children element should contain correct element", () => {
        const component = create(<HeadLine><h1>Hello</h1></HeadLine>)
        const root = component.root;

        expect(root.props.children.type).toBe("h1");
    });
})