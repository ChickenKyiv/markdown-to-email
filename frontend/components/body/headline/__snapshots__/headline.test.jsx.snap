// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headline/index component renders correctly 1`] = `
<h1>
  Some text
</h1>
`;
